src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java

```diff
@@ -223,4 +230,21 @@ private void removeEdge(final Edge<V> edge) {
         internalGraph.removeEdge(unpackEdge(edge));
     }
 
+    /**
+     * @author Rutger van den Berg
+     *         Compares two edges by their target vertex.
+     */
+    class EdgeComparatorByVertex implements Comparator<Edge<V>> {
```

Comment:
The main difference in readability between your example and the one I use is that you switched the full if to an inline conditional. Not so much lambda vs inner class. 

Further, my interpretation of the presentation you linked to is that lambdas are sometimes faster/can be faster, but are sometimes slower as well. 

Conclusion - I see no reason to change this.